### DISCUSSION TITLE
How to use externalsType: script, as module

### DISCUSSION THREAD

**Discussion by felipe-chamas** [webpack/webpack#12860] [Q&A] [answered]
I'm trying to import `@google/model-viewer` a CDN dependency which requires the following sintax:

```html
<script type="module" src="https://unpkg.com/@google/model-viewer@1.5.0/dist/model-viewer.min.js" />
```

Is there a way to set this type? My current configuration throws the following error: `Uncaught SyntaxError: export declarations may only appear at top level of a module -> Uncaught (in promise) ScriptExternalLoadError: Loading script failed.`
```js
// webpack.config.js
module.exports = {
    ...
    externalsType: 'script',
    externals: {
      '@google/model-viewer':
        'global@https://unpkg.com/@google/model-viewer@1.5.0/dist/model-viewer.min.js',
    }
}
```
 Looking at the code [here](https://github.com/webpack/webpack/blob/80c5041fe1f909d34975abd33e03bdb069876ea9/lib/runtime/LoadScriptRuntimeModule.js#L68~L91) I think it's not an option...

**Comment by sokra** [comment, marked as answer]
Script externals are for classic script. For modules you can use `import()`:

``` js
​// webpack.config.js​
​module​.​exports​ ​=​ ​{​
    ...
    ​externalsType​: ​'import'​,​
    ​externals​: ​{​
      ​'@google/model-viewer'​:
        ​'https://unpkg.com/@google/model-viewer@1.5.0/dist/model-viewer.min.js'​,​
    ​}​
​}
```

**Reply by felipe-chamas** [reply]
Now I'm getting an error at build time:
`Error: The target environment doesn't support 'import()' so it's not possible to use external type 'import'`

**Reply by felipe-chamas** [reply]
I'm also importing `import '@google/model-viewer'` in my code, just to clarify

**Reply by sokra** [reply]
If you get this error your browserslist include browsers that doesn't support `import()`, or you don't have one at all.

**Reply by sokra** [reply]
You need to use https://webpack.js.org/configuration/output/#outputimportfunctionname and provide a polyfill

**Reply by felipe-chamas** [reply]
All right, thank you!!!
